cartofacade/capi_test.go

```diff
@@ -126,42 +86,47 @@ func TestCGoAPI(t *testing.T) {
 		test.That(t, err, test.ShouldBeNil)
 		test.That(t, pvcl, test.ShouldNotBeNil)
 
-		cfg := getBadTestConfig()
-		algoCfg := getTestAlgoConfig()
-		vc, err := New(cfg, algoCfg, pvcl)
+		cfgBad := GetBadTestConfig()
+		cfg, dir, err := GetTestConfig("mysensor")
+		algoCfg := GetTestAlgoConfig()
+		vc, err := NewCarto(cfg, algoCfg, &CartoLibMock{})
+
+		// initialize viam_carto with an invalid library incorrectly
+		test.That(t, err, test.ShouldResemble, errors.New("cannot cast provided library to a CartoLib"))
```

Comment:
@nicksanford added in this test, lmk if It looks good to you and I will merge this PR in!

Reply:
looks great now, go ahead & merge